tracing-subscriber/src/fmt/span.rs

```diff
@@ -62,8 +63,56 @@ enum State {
     Empty(usize),
 }
 
+struct SpanStackElement {
+    id: Id,
+    duplicate: bool,
+}
+
+struct SpanStack {
+    stack: Vec<SpanStackElement>,
+    ids: HashSet<Id>,
+}
+
+impl SpanStack {
+    fn new() -> Self {
+        SpanStack {
+            stack: vec![],
+            ids: HashSet::new(),
+        }
+    }
+
+    fn push(&mut self, id: Id) {
+        let duplicate = self.ids.contains(&id);
+        if !duplicate {
+            self.ids.insert(id.clone());
+        }
+        self.stack.push(SpanStackElement { id, duplicate })
+    }
+
+    fn pop(&mut self, expected_id: &Id) -> Option<Id> {
+        if self.stack.last().map(|elt| &elt.id) == Some(expected_id) {
+            let SpanStackElement { id, duplicate } = self.stack.pop()?;
+            if !duplicate {
+                self.ids.remove(&id);
+            }
+            Some(id)
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    fn find_last_no_duplicate(&self) -> Option<&Id> {
```

Comment:
Nit, take it or leave it: I might have just called this `current` --- IMO that describes the _semantic_ meaning of the method, rather than what it _does_...

Reply:
ok!